usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateNoOverlappingStopTimeInTripBlock.java

```diff
@@ -0,0 +1,264 @@
+                                    // This is done by checking files `calendar.txt` and `calendar_dates.txt
+
+                                    // | routeId | tripId | serviceId | blockId | first stop time | last stop time |
+                                    // |---------|--------|-----------|---------|-----------------|----------------|
+                                    // | 0       | 2      | a         | 5       | 07:00           | 10:00          |
+                                    // | 0       | 5      | b         | 5       | 08:45           | 11:20          |
+                                    // | 0       | 8      | c         | 5       | 11:40           | 13:50          |
+                                    Calendar currentTripCalendar = null;
+                                    Calendar unvisitedTripCalendar = null;
+                                    Map<String, CalendarDate> currentTripCalendarDatePossibilityCollection = null;
+                                    Map<String, CalendarDate> unvisitedTripCalendarDatePossibilityCollection = null;
+                                    if (areCalendarProvided) {
+                                        currentTripCalendar =
+                                                dataRepo.getCalendarByServiceId(currentTrip.getServiceId());
+                                        unvisitedTripCalendar =
+                                                dataRepo.getCalendarByServiceId(unvisitedTrip.getServiceId());
+                                    } else {
+                                        currentTripCalendarDatePossibilityCollection =
+                                                dataRepo.getCalendarDateAll().get(currentTrip.getServiceId());
+                                        unvisitedTripCalendarDatePossibilityCollection =
+                                                dataRepo.getCalendarDateAll().get(unvisitedTrip.getServiceId());
+                                    }
+                                    // interpret data from `calendar.txt` to determine if currentTrip and unvisitedTrip
+                                    // operate on same days
+                                    if (areCalendarProvided) {
+                                        // first check if `start_date` and `end_date` overlap for both trips
+                                        if (currentTripCalendar.areCalendarOverlapping(unvisitedTripCalendar)) {
+                                            // determine the if the trips operate on same days
+                                            if (currentTripCalendar.getCalendarsCommonOperationDayCollection(
+                                                    unvisitedTripCalendar).size() >= 1) {
+
+                                                if (unvisitedTripFirstTime != null && unvisitedTripLastTime != null) {
+
+                                                    if (unvisitedTripFirstTime <= unvisitedTripLastTime) {
```

Comment:
Same as previous - The trip first and last time can't be equal, right? Otherwise the trip is 0 seconds long?

Also, if this is another sanity check, I'd suggest refactoring to areTripTimesValid() method to better explain what's happening.